hydra-main/src/main/java/com/addthis/hydra/job/spawn/balancer/SpawnBalancer.java

```diff
@@ -1185,14 +1213,46 @@ protected void updateAggregateStatistics(Iterable<HostState> hosts) {
             updateActiveJobIDs();
             double maxMeanActive = -1;
             double maxDiskPercentUsed = -1;
+            long maxDiskFree = 0;
+            long minDiskFree = Long.MAX_VALUE;
+            long sumDiskFree = 0;
+            double maxTaskPercent = 0;
+            double minTaskPercent = 1;
+            double sumTaskPercent = 0;
             for (HostState host : hosts) {
                 maxMeanActive = Math.max(maxMeanActive, host.getMeanActiveTasks());
                 maxDiskPercentUsed = Math.max(maxDiskPercentUsed, getUsedDiskPercent(host));
             }
             for (HostState host : hosts) {
-                cachedHostScores.put(host.getHostUuid(),
-                                     calculateHostScore(host, maxMeanActive, maxDiskPercentUsed));
+                // metrics here
+                HostScore score = calculateHostScore(host, maxMeanActive, maxDiskPercentUsed);
+                cachedHostScores.put(host.getHostUuid(), score);
+                long freeDisk = score.getFreeDiskBytes();
+                double taskPercent = score.getScoreValue(false);
+                sumDiskFree += freeDisk;
+                sumTaskPercent += taskPercent;
+                maxDiskFree = Math.max(freeDisk, maxDiskFree);
+                minDiskFree = Math.min(freeDisk, minDiskFree);
+                maxTaskPercent = Math.max(taskPercent, maxTaskPercent);
+                minTaskPercent = Math.min(taskPercent, minTaskPercent);
             }
+            // update average metrics
+            int numScores = cachedHostScores.size();
+            long avgDiskFree = sumDiskFree / (long) numScores;
+            double avgTaskPercent = sumTaskPercent / (double) numScores;
+            long sumDiskFreeDiff = 0;
+            double sumTaskPercentDiff = 0;
+            for (HostScore score : cachedHostScores.values()) {
+                long diskDiff = Math.abs(avgDiskFree - score.getFreeDiskBytes());
+                sumDiskFreeDiff += diskDiff;
+                sumTaskPercentDiff += Math.abs(avgTaskPercent - score.getScoreValue(false));
+            }
+            avgDiskFreeDiff = sumDiskFreeDiff / (long) numScores;
+            avgTaskPercentDiff = sumTaskPercentDiff / (double) numScores;
+            minDiskFreeDiff = avgDiskFree - minDiskFree;
+            maxDiskFreeDiff = maxDiskFree - avgDiskFree;
```

Comment:
Typically when you are writing scores for a scheduler you want to normalize the variables so your scheduler is fair. 

I.e. 
`(total - free) / total`

The current way you are keeping track of things works fine when your cluster is homogeneous, but it will start to be unfair when nodes of different sizes enter the picture.